cognite/client/data_classes/simulators/routine_revisions.py

```diff
@@ -0,0 +1,653 @@
+        return super().dump(camel_case=camel_case)
+
+
+@dataclass
+class SimulatorRoutineSteadyStateDetection(CogniteObject):
+    """
+    Steady State Detection checks for steady state regions in a given time series.
+    The user specifies the time series and three parameters: min section size, var threshold, and slope threshold.
+    It returns a binary time series, with 1 for timestamps where the steady state criteria is met and 0 otherwise.
+
+    Args:
+        enabled (bool): Whether the steady state detection is enabled.
+        timeseries_external_id (str): The external ID of the time series to check for steady state.
+        aggregate (str): The aggregation method to use for the time series.
+        min_section_size (int): The minimum number of consecutive data points that must meet the steady state criteria.
+        var_threshold (float): The maximum variance allowed for the steady state region.
+        slope_threshold (float): The maximum slope allowed for the steady state region.
+    """
+
+    enabled: bool = False
+    timeseries_external_id: str | None = None
+    aggregate: str | None = None
+    min_section_size: int | None = None
+    var_threshold: float | None = None
+    slope_threshold: float | None = None
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        instance = super()._load(resource, cognite_client)
+        return instance
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        return super().dump(camel_case=camel_case)
+
+
+@dataclass
+class SimulatorRoutineConfiguration(CogniteObject):
+    """
+    The simulator routine configuration defines the configuration of a simulator routine revision.
+    Learn more about simulator routine configuration <https://docs.cognite.com/cdf/integration/guides/simulators/simulator_routines>.
+
+    Args:
+        schedule (SimulatorRoutineSchedule): Schedule configuration.
+        data_sampling (SimulatorRoutineDataSampling): Data sampling configuration. Learn more about data sampling <https://docs.cognite.com/cdf/integration/guides/simulators/about_data_sampling>.
+        logical_check (list[SimulatorRoutineLogicalCheck]): Logical check configuration.
+        steady_state_detection (list[SimulatorRoutineSteadyStateDetection]): Steady state detection configuration.
+        inputs (list[SimulatorRoutineInputConstant | SimulatorRoutineInputTimeseries] | None): The inputs of the simulator routine revision.
+        outputs (list[SimulatorRoutineOutput] | None): The outputs of the simulator routine revision.
+    """
+
+    schedule: SimulatorRoutineSchedule
+    data_sampling: SimulatorRoutineDataSampling
+    logical_check: list[SimulatorRoutineLogicalCheck]
+    steady_state_detection: list[SimulatorRoutineSteadyStateDetection]
+    inputs: list[SimulatorRoutineInputConstant | SimulatorRoutineInputTimeseries] | None
```

Comment:
This is used in the write format, so I think we should use the default arg and not require the user to specify it.
```suggestion
    inputs: list[SimulatorRoutineInputConstant | SimulatorRoutineInputTimeseries] | None = None
```

Reply:
Got it, then it is fine